wpilibj/src/main/java/edu/wpi/first/wpilibj/GenericHID.java

```diff
@@ -283,4 +285,15 @@ public void setRumble(RumbleType type, double value) {
     }
     HAL.setJoystickOutputs((byte) m_port, m_outputs, m_leftRumble, m_rightRumble);
   }
+
+  /**
+   * Returns a {@link Button} object corresponding to the specified button number from this HID.
+   * Note: this returns a Button from the frc2 Command API, not from the legacy command API.
+   *
+   * @param button the number of the button to return (starting at 1)
+   * @return the {@link Button} object
+   */
+  public Button getButton(int button) {
+    return new JoystickButton(this, button);
+  }
```

Comment:
Just like in my comment in the C++ variant of this, I don't like how this breaks encapsulation and creates a dependency from a fundamental class back to the command stuff.  I would like to find a way to architect this so that this is not necessary.

Reply:
It's not strictly necessary, but it removes a fair bit of verbosity (especially in conjunction with the new button composition) and the coupling is not very major (none of the actual workings of the button class are changed or depend on the command framework); several users told me they'd find this method convenient.